osquery/tables/system/python_packages.cpp

```diff
@@ -84,12 +84,43 @@ void genPackage(const std::string& path, Row& r, Logger& logger) {
   }
 }
 
+Status stdListDirectoriesInDirectory(const fs::path& path,
+                                     std::vector<std::string>& results,
+                                     bool recursive) {
+  if (path.empty() || !pathExists(path)) {
+    return Status(1, "Target directory is invalid");
+  }
+
+  if (recursive) {
+    for (const auto& entry : fs::recursive_directory_iterator(path)) {
+      if (fs::is_directory(entry)) {
+        results.push_back(entry.path().string());
+      }
+    }
+  } else {
+    for (const auto& entry : fs::directory_iterator(path)) {
+      if (fs::is_directory(entry)) {
+        results.push_back(entry.path().string());
+      }
+    }
+  }
+
+  return Status::success();
+}
+
 void genSiteDirectories(const std::string& site,
                         QueryData& results,
                         Logger& logger) {
   std::vector<std::string> directories;
-  if (!listDirectoriesInDirectory(site, directories, true).ok()) {
-    return;
+
+  if (isPlatform(PlatformType::TYPE_WINDOWS)) {
```

Comment:
Is this a change that we should apply on all platforms? Is there something windows specific about it?

Reply:
Yes, the` listDirectoriesinDirectory` helper takes a long time to complete on Windows. This is a field issue reported by Fleet customers - see [here](https://github.com/fleetdm/fleet/issues/8243). I decided to play safe and just use the new implementation on Windows only.